File: Selenium/Activities/Activity17.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

import java.time.Duration;
import java.util.Iterator;
import java.util.List;

public class Activity17 {
    public static void main(String[] args) throws InterruptedException {
        //Setup GekoDriver
        WebDriverManager.firefoxdriver().setup();

        //Create driver Object for Firefox Browser
        FirefoxDriver driver = new FirefoxDriver();

        //Set Implicit wait of 5 Secs
        driver.manage().timeouts().implicitlyWait(Duration.ofMillis(5000));
        //Open the url
        driver.get("https://v1.training-support.net/selenium/selects");

        //Get the Title
        String title = driver.getTitle();
        System.out.print("Title of the Page is : " + title + "\n");

        //Get the Handle of Dropdown
        WebElement dropDown = driver.findElement(By.xpath("//select[@id='single-select']"));
        Select dropDownHandle = new Select(dropDown);

        //Select 2nd Option by Visible Text
        Thread.sleep(2000);
        dropDownHandle.selectByVisibleText("Option 2");

        //Select 2nd Option by Value
        Thread.sleep(5000);
        dropDownHandle.selectByValue("4");

        //Get the Options from the Dropdown into a List and Print the items
        List<WebElement> items = dropDownHandle.getOptions();
        for (WebElement item : items){
            System.out.println(item.getText());
        }

        //Wait for 5 Seconds and Close the Browser
        Thread.sleep(5000);
        driver.quit();
    }
}

File: Selenium/Activities/Activity17.py
# Import required packages from selenium
import time

from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.select import Select
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver using WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Set Explicit wait of 3 Secs
    wait = WebDriverWait(driver, 10)

    # Launch the URL and get the title
    driver.get("https://v1.training-support.net/selenium/selects")
    home_title = driver.title
    print("Title of the Home Page is : {}".format(home_title))

    # Get the Handle of Dropdown
    dropDown = driver.find_element(By.XPATH, "//select[@id='single-select']")
    dropDownHandle = Select(dropDown)

    # Select 2nd Option by Visible Text
    time.sleep(3)
    dropDownHandle.select_by_visible_text("Option 2")

    # Select 2nd Option by Value
    time.sleep(3)
    dropDownHandle.select_by_value("4")

    # Get the Options from the Dropdown into a List and Print the items
    items = dropDownHandle.options
    for item in items:
        print(item.text)

    # Wait for 3 Seconds and Close the Browser
    time.sleep(3)
    driver.close()
